app/controllers/ValidationTaskController.scala

```diff
@@ -0,0 +1,160 @@
+        val labelMetadataJson: JsObject = LabelTable.validationLabelMetadataToJson(labelMetadata)
+        Future.successful(Ok(labelMetadataJson))
+      case _ => Future.successful(Ok(Json.obj("error" -> "no such label")))
+    }
+  }
+
+  /**
+    * This gets a random list of labels to validate for this mission.
+    * @param count  Number of labels to retrieve for this list.
+    * @return       JsValue containing a list of labels with the following attributes:
+    *               {label_id, label_type, gsv_panorama_id, heading, pitch, zoom, canvas_x, canvas_y,
+    *               canvas_width, canvas_height}
+    */
+  def getLabelListForValidation(count: Int): JsValue = {
+    val labelMetadata: Seq[LabelValidationMetadata] = LabelTable.retrieveRandomLabelListForValidation(count)
+    val labelMetadataJsonSeq: Seq[JsObject] = labelMetadata.map(label => LabelTable.validationLabelMetadataToJson(label))
+    val labelMetadataJson : JsValue = Json.toJson(labelMetadataJsonSeq)
+    labelMetadataJson
+  }
+
+  /**
+    * This function gets the metadata for a random label in the database.
+    * @return Label metadata containing GSV metadata and label type
+    */
+  def getRandomLabelData = UserAwareAction.async { implicit request =>
+    val labelMetadata: LabelValidationMetadata = LabelTable.retrieveSingleRandomLabelForValidation()
+    val labelMetadataJson: JsObject = LabelTable.validationLabelMetadataToJson(labelMetadata)
+    Future.successful(Ok(labelMetadataJson))
+  }
+
+  /**
+    *
+    * @param user
+    * @param missionProgress  Metadata for this mission
+    * @return
+    */
+  def updateMissionTable(user: Option[User], missionProgress: ValidationMissionProgress): Option[Mission] = {
+    val missionId: Int = missionProgress.missionId
+    val skipped: Boolean = missionProgress.skipped
+    val userId: UUID = user.get.userId
+    val role: String = user.get.role.getOrElse("")
+
+    //  (missionProgress.labelsProgress.isEmpty) Logger.error("Received null labels progress for validation mission.")
+    val labelsProgress: Int = missionProgress.labelsProgress
+    println("[ValidationTaskController] updateMissionTable: " + missionId + ", " + skipped + ", " + userId + ", " + role);
+
+    if (missionProgress.completed) {
+      println("[ValidationTaskController] updateMissionTable: This mission is complete")
+      // TODO: replace 0.0 with pay per label (later)
+      MissionTable.updateCompleteAndGetNextValidationMission(userId, 0.0, missionId, labelsProgress, skipped)
+    } else {
+      println("[ValidationTaskController] updateMissionTable: This mission is not complete")
+      MissionTable.updateValidationProgressOnly(userId, missionId, labelsProgress)
+    }
+  }
+
+  def getLabelList(missionProgress: ValidationMissionProgress): Option[JsValue] = {
+    // Retrieve more labels. Currently hard-coded to be 10 labels.
+    // TODO: Get rid of hard-coded 10 value.
```

Comment:
Adding a comment to help you remember you had this TODO comment.

Reply:
This is fixed!